pyrokinetics/gk_code/GKOutputReaderGENE.py

```diff
@@ -80,7 +83,8 @@ def infer_path_from_input_file(filename: PathLike) -> Path:
         # the directory.
         filename = Path(filename)
         num_part_regex = re.compile(r"(\d{4})")
-        num_part_match = num_part_regex.search(filename.name)
+        num_part_match = num_part_regex.search(str(Path(filename).name))
```

Comment:
Were you getting an error with some input here? `filename` should already be a `Path` from L84 above, and I thought that `Path.name` was already `str`

Reply:
I can no longer reproduce the error. Maybe when I originally forked from `unstable`, L84 was not there and appeared after a later merge with `unstable`. I will remove this change. Thank you for noticing it!